### ISSUE TITLE
Delta doesn't start on Windows 7

### ISSUE THREAD

**Issue by missytake** [deltachat/deltachat-desktop#1323]
- Platform (linux/mac/windows): windows 7
- Delta Chat Desktop Version: 0.901.0 (setup.exe)
- Expected behavior: Delta Chat starts
- Actual behavior: The process exits after a few seconds.
- Steps to reproduce the problem: Start the desktop client (no matter which way)
- Screenshots: 

![windows screenshot](https://oc2.links-tech.org/s/R4fWFn6rE1FG39h/download)

- Logs: 

console output:

```
C:\Users\mike>AppData\Local\Programs\deltachat-desktop\DeltaChat.exe

C:\Users\mike>
 Logfile: C:\Users\mike\AppData\Local\DeltaChat\logs\2020-01-08-19-08-11.log.tsv

C:\Users\mike\AppData\Local\Programs\deltachat-desktop\resources\app.asar.unpacked\node_modules\deltachat-node\prebuilds\win32-x64\node.napi.node
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:160:31)

    at Object.Module._extensions..node (internal/modules/cjs/loader.js:722:18)
    at Object.module.(anonymous function) [as .node] (as .node] (ELECTRON_ASAR.js:169:18)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at load (C:\Users\mike\AppData\Local\Programs\deltachat-desktop\resources}app.asar\mode_modules\node-gyp-build\index.js:20:10)
    at Object.<anonymous> (C:\Users\mike\AppData\Local\Programs\deltachat-desktop\resources\app.asar\node_modules\deltachat-node\binding.js:1:198)
```

the logfile at C:\Users\mike\AppData\Local\DeltaChat\logs\2020-01-08-19-08-11.log.tsv is empty.

**Comment by Jikstra** [comment]
@missytake  @gerryfrancis  Can you try out this portable build from this pr https://github.com/deltachat/deltachat-desktop/pull/1349 if this bug still happens?

https://download.delta.chat/desktop/preview/deltachat-desktop-1349.exe
Mirror in case pr got merged:
https://transfer.sh/b8FRj/deltachat-desktop-1349.exe

The portable build will not find your existing configuration so you need to setup again.

**Comment by missytake** [comment]
Unfortunately it still doesn't start. (Can't say anything about the portable folder.)

I don't know if it is still the same traceback, as the traceback only appears with setup.exe versions o.0

**Comment by Jikstra** [comment]
Do you run it in a vm? The portable.exe is heavily compressed so it can take some time until it's unpacked.

**Comment by gerryfrancis** [comment]
@Jikstra Actually I have not tested Delta Chat on Windows 7, but I think I can grab an appropriate PC system to find out. (So far, Delta Chat runs just fine with Windows 8.1 Pro.) I have thought that the "deltachat.dll" file might not be installed on Windows 7 (properly). I will check that as soon as I can.
@missytake Could your antivirus software block "deltachat.dll"?

**Comment by Axehaft** [comment]
I might be having the same issue on a Windows 10 machine.  When I try to start Delta, I get the little blue circle for about a second and then nothing.  Nothing opens.  I'm not sure how to get log files, but willing to help.  I'm on a Microsoft Surface: 
![surface](https://user-images.githubusercontent.com/46401365/72532557-48916180-3839-11ea-88cc-3ffd2c949482.png)

**Comment by gerryfrancis** [comment]
@Axehaft Have you installed Delta Chat by running setup.exe before, or are you trying to run the portable release?

**Comment by Axehaft** [comment]
I used the setup file:  DeltaChat Setup 0.901.0

**Comment by gerryfrancis** [comment]
@Axehaft And have you installed Delta Chat already but the app itself does not run? Or does the setup itself not start?

**Comment by Axehaft** [comment]
Ahhh sorry.  I was not clear.  I installed Delta Chat.  Everything shows that it installed OK.  When I click on the shortcut for the app itself, that is when I get the blue circle for a second and then nothing.

**Comment by gerryfrancis** [comment]
@Axehaft Never mind. Please check the recent log file located in path "C:\Users\YourUserProfile\AppData\Local\DeltaChat\logs". ("YourUserProfile" is a placeholder for the name of your Windows user profile.) If there is a crash, it should be logged there.

**Comment by Axehaft** [comment]
A log is created, but it is empty. 
![log](https://user-images.githubusercontent.com/46401365/72535827-d28ff900-383e-11ea-9568-74b7377f6aa1.png)
When I open the file, there is nothing there.

**Comment by gerryfrancis** [comment]
Please look for entries of the source "Application Error" in the Application log (by running Windows Event Viewer).

**Comment by Axehaft** [comment]
Not seeing anything show up there.  This is what I'm seeing ... maybe I'm not looking in the right place?
![log_001](https://user-images.githubusercontent.com/46401365/72538733-e1c57580-3843-11ea-8308-43b16457ed62.png)

**Comment by missytake** [comment]
@Jikstra 

You requested further information on my Windows version, path to .exe, and username in a private chat:

* Windows Version 6.1 (Build 7600)
* .exe Path: `C:\Users\mike\Downloads\DeltaChat0.901.0.exe`
* Username: mike

**Comment by gerryfrancis** [comment]
@Axehaft The place is correct. Actually I am out of ideas... :/

**Comment by missytake** [comment]
@Axehaft You are running the setup version, not the portable version, right?

Can you open cmd.exe (with the windows+R key combination), and execute `AppData\Local\Programs\deltachat-desktop\DeltaChat.exe`? This gave me a traceback earlier when I tried it.

**Comment by Axehaft** [comment]
@missytake Cool!  Thanks.  Looks like the same as what you had: 
![log_002](https://user-images.githubusercontent.com/46401365/72540661-c7d96200-3846-11ea-9426-0ff5204adde5.png)

**Comment by gerryfrancis** [comment]
@Axehaft @missytake Please check if the file "deltachat.dll" is present in "%USERPROFILE%\AppData\Local\Programs\deltachat-desktop\resources\app.asar.unpacked\node_modules\deltachat-node\prebuilds\win32-x64". It should be there. If it is not, probably an antivirus software might have scanned the file, detected a false positive, and maybe deleted the file.

**Comment by missytake** [comment]
It's there, and I don't have antivirus software

**Comment by Axehaft** [comment]
deletachat.dll is there for me too.  I don't have antivirus software and I disabled Windows Defender.

**Comment by gerryfrancis** [comment]
Thanks for reporting. At least we can rule out antivirus software, and the issue is not limited to Windows 7...

**Comment by Axehaft** [comment]
One more thing.  The moduel that the log says can't be found seems to be there:
![log_003](https://user-images.githubusercontent.com/46401365/72543579-94e59d00-384b-11ea-8351-28e05e4c86bb.png)

**Comment by missytake** [comment]
> @missytake @gerryfrancis Can you try out this portable build from this pr #1349 if this bug still happens?
> 
> https://download.delta.chat/desktop/preview/deltachat-desktop-1349.exe
> Mirror in case pr got merged:
> https://transfer.sh/b8FRj/deltachat-desktop-1349.exe
> 
> The portable build will not find your existing configuration so you need to setup again.

After a Windows Update which also fixed my C++ Redistributable issue, this build suddenly works for me. Seems it was a problem with my Windows installation... @Axehaft can you install the latest Windows Updates and try again?

**Comment by Axehaft** [comment]
Ok.  Everything is completely up to date on my Surface, still no go = little blue wheel for a second or so and then nothing.  I deleted and reinstalled again as well.

**Comment by Simon-Laux** [comment]
_A friend installed it on newest win10 64bit and it worked fine for them._

**Comment by Axehaft** [comment]
Yes.  I works fine on my wifes laptop and my desktop machines, all which are Windows 10.  Just not my Microsoft Surface which is Windows 10 Pro.  Without logs. there seems to be no way of knowing why it wont work.

**Comment by gerryfrancis** [comment]
@Axehaft The Surface models feature displays with very high resolutions. Which do you run on your three systems? (The Surface might have the highest one.) (Just a shot in the dark...)

**Comment by Simon-Laux** [comment]
Can you open the exe with cmd.exe or powershell? maybe you get an error in there

**Comment by Axehaft** [comment]
@Simon-Laux Already did that.  Posted above: https://github.com/deltachat/deltachat-desktop/issues/1323#issuecomment-575218988

**Comment by Jikstra** [comment]
@Axehaft, can you extract the mentioned app.asar? I think 7zip can do it, otherwise there's an npm module for it too. I would like to know if the `node_modules/deltachat-node/prebuilds/win32-x64/node.napi.node` is actually in that archive or not.

**Comment by Axehaft** [comment]
@Jikstra I think I understand what you want me to do, but I need a bit more instruction how to do it.  I need to open something with 7zip ... what file should I open?

**Comment by Jikstra** [comment]
I just came accross a very similiar error when working on deltachat-node on windows. Basically what can cause this error is that some dependencies (read some `.dll` files) are missing. One way to figure out WHICH .dll are missing, 

Instruction:
1. locate the node.napi.node file (@Axehaft you have it in the screenshot here https://github.com/deltachat/deltachat-desktop/issues/1323#issuecomment-575234892)
2. Download the dependencies tool. It lists all .dll files required by a file. You can download it here: https://github.com/lucasg/Dependencies/releases
3. Extract the .zip and run the DependenciesGui.exe file.
4. Go to file -> open -> select the node.napi.node file
5. Report back the output of the tool. We can also do an audio/video call to sort this out :)

**Comment by gerryfrancis** [comment]
The same behavior occurs when the file "vcruntime140.dll", which is part of the "Microsoft Visual C++ 2015-2019 Redistributable", is not present in the path "C:\Windows\System32`\".
@Axehaft Could you please check if the file "vcruntime140.dll" exists in "C:\Windows\System32\"?

**Comment by Jikstra** [comment]
@Axehaft will close this issue for now, please reopen it if it's still happening to you with the latest desktop version (0.999.1)

**Comment by TheOtherDoctor** [comment]
Having the very same problem with installing version 1.1.1 on Windows 10: install works smooth, then the stopping with missing module,
```
Error: The specified module could not be found.
\\?\C:\Users\xxxxxxx\AppData\Local\Programs\deltachat-desktop\resources\app.asar.unpacked\node_modules\deltachat-node\prebuilds\win32-x64\node.napi.node
    at process.func (electron/js2c/asar.js:138:31)
    at process.func [as dlopen] (electron/js2c/asar.js:138:31)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:828:18)
    at Object.func (electron/js2c/asar.js:138:31)
    at Object.func [as .node] (electron/js2c/asar.js:147:18)
    at Module.load (internal/modules/cjs/loader.js:645:32)
    at Function.Module._load (internal/modules/cjs/loader.js:560:12)
    at Module.require (internal/modules/cjs/loader.js:685:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at load (C:\Users\xxxxxxx\AppData\Local\Programs\deltachat-desktop\resources\app.asar\node_modules\node-gyp-build\index.js:21:10)
```
Could not determine from the comments above how to actively solve this. 
Did I miss some prerequisites which need to be installed?

Tried the previews 1495 and 1472 - there not even the install process seems to start. :-(
Did not try the portable version yet.

**Comment by Simon-Laux** [comment]
Do you have a 32 or 64bit system?

Am 25. März 2020 02:16:17 MEZ schrieb TheOtherDoctor <notifications@github.com>:
>Having the very same problem with installing version 1.1.1 on Windows
>10: install works smooth, then the stopping with missing module,
>```
>Error: The specified module could not be found.
>\\?\C:\Users\xxxxxxx\AppData\Local\Programs\deltachat-desktop\resources\app.asar.unpacked\node_modules\deltachat-node\prebuilds\win32-x64\node.napi.node
>    at process.func (electron/js2c/asar.js:138:31)
>    at process.func [as dlopen] (electron/js2c/asar.js:138:31)
>at Object.Module._extensions..node
>(internal/modules/cjs/loader.js:828:18)
>    at Object.func (electron/js2c/asar.js:138:31)
>    at Object.func [as .node] (electron/js2c/asar.js:147:18)
>    at Module.load (internal/modules/cjs/loader.js:645:32)
>    at Function.Module._load (internal/modules/cjs/loader.js:560:12)
>    at Module.require (internal/modules/cjs/loader.js:685:19)
>    at require (internal/modules/cjs/helpers.js:16:16)
>at load
>(C:\Users\xxxxxxx\AppData\Local\Programs\deltachat-desktop\resources\app.asar\node_modules\node-gyp-build\index.js:21:10)
>```
>Could not determine from the comments above how to actively solve this.
>
>Did I miss some prerequisites which need to be installed?
>
>Tried the previews 1495 and 1472 - there not even the install process
>seems to start. :-(
>Did not try the portable version yet.
>
>
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/deltachat/deltachat-desktop/issues/1323#issuecomment-603586644

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by TheOtherDoctor** [comment]
64bit. Win 10 home, Version 1909, Build 18363.720, on a Lenovo T400.
No devel tools or the like installed, as I am preparing a rollout for a non-techie person.

**Comment by TheOtherDoctor** [comment]
I installed the mentioned Dependencies tool, opened the file - and cannot understand what I see. :-(

The vcruntime, which has also been mentioned above, "vcruntime140.dll" exists in "C:\Windows\System32", is called vcruntime140_clr0400.dll on my system.

Anything else I could do?

**Comment by Jikstra** [comment]
@TheOtherDoctor, can you you give us a screenshot from the output of the dependencies tool :)

**Comment by TheOtherDoctor** [comment]
So this is the file I throw into the DepTool...
![DepTool_filepick](https://user-images.githubusercontent.com/62629463/78078056-c8938600-73a9-11ea-94a7-d9f52eada19b.png)
...and this is what I see then:
![DepTool_filedeps](https://user-images.githubusercontent.com/62629463/78078155-ed87f900-73a9-11ea-9262-5d65efdcfe56.png)

As mentioned before, the vcruntime has a different name, I guess that means it does not match the requirement and I'd have to install some vc_redist separately?

**Comment by Jikstra** [comment]
> As mentioned before, the vcruntime has a different name, I guess that means it does not match the requirement and I'd have to install some vc_redist separately?

I don't know too much about windows, but I guess this is the problem. We need to try it out and see if it helps. If it helps it's maybe possible to bundle that .dll with deltachat-desktop, but that would be a next step. 


> Remaining question: I have to assert this manually whenever I install the desktop app on a Win10 PC, or should that have been handled during installation?

I know that it's sometimes possible to just put the .dll next to the .exe (maybe also next to the .node file) and then it magically finds it. But it could be possible that there will be legal problems with this and we would somehow need to make sure that the visual c runtime is getting installed along deltachat desktop. 

Thank you very much for figuring this out!

**Comment by TheOtherDoctor** [comment]
You're welcome - and thanks a lot for all _your_ work! 
About making the app install smoother on Windows (which I have no significant idea about, as a devel platform), before investing work into fixing this specific issue, I'd propose to consider making any such error visible for the user. 
Even something as simple as some try/catch-like post-fail splash screen "app failed to start, for troubleshooting please have a look into https://... (README) ..." would be already way better than silently dying.

**Comment by gerryfrancis** [comment]
IMHO, that explains why Delta Chat runs out of the box on most Windows systems. Nevertheless, the installer of Delta Chat should include Visual C++ Runtime from Microsoft, and check whether it is necessary to install it. Windows does not feature Visual C++ Runtime by default.

**Comment by hpk42** [comment]
We just got another user who remotely helped a friend set up DC on windows -- they had to install the Visual C++ runtime as well.

**Comment by Axehaft** [comment]
Hello All.  I just want to add that I also installed the Visual C++ Runtime from Microsoft ... and now DC starts and runs well on my Surface.

**Comment by Simon-Laux** [comment]
Hey folks, I have a portable version for you to test https://testrun.org/DeltaChat%201.4.0.testing.exe
It works for me on my windows vm, but I want to know if it works for you too.

**Comment by gerryfrancis** [comment]
@Simon-Laux Thanks a lot, but unfortunately I cannot run this file on my system (Windows 8.1 Pro):

![grafik](https://user-images.githubusercontent.com/51095533/85366219-32de4180-b527-11ea-821e-34f6d939f172.png)

**Comment by gerryfrancis** [comment]
@Simon-Laux While this version works fine on my Windows system, I wanted to test the prior version 1.3.4 (without Visual C++ runtime) to see whether it works, too, without the runtime. For this purpose, I uninstalled Visual C++ runtime 2017, 2015, and 2013 - but 1.3.4 still works fine! What am I doing wrong??

**Comment by gerryfrancis** [comment]
@Simon-Laux It seems some other software product(s) want(s) to keep the runtime dll files on my Windows system. (I have found them in the Windows system folders, although I officially uninstalled these runtimes.) Since I do not have another system I could test the new version on, I am afraid to say that I cannot continue to check, sorry for that. :(

**Comment by Simon-Laux** [comment]
@gerryfrancis no problem.

**Comment by missytake** [comment]
> Hey folks, I have a portable version for you to test https://testrun.org/DeltaChat%201.4.0.testing.exe

That version seems to work on my machine :) great. This means we can close this issue, right?

**Comment by Darthagnon** [comment]
The last )version of DeltaChat to run on Windows 7 is [DeltaChat v1.40.4](https://github.com/deltachat/deltachat-desktop/releases/tag/v1.40.4). Corporate sponsorship for Windows 7 builds was available until v1.42.2. Since then, due to Electron version bumps, Deltachat does not run on Windows 7. 

However, recently I saw that Deltachat has Tauri builds available. [Tauri should work on Windows 7](https://v2.tauri.app/distribute/windows-installer/#supporting-windows-7), provided [rust is built with v1.77.2 or the Windows 7 build  arguments](https://github.com/tauri-apps/tauri/discussions/12016). Webview2 v109.xx.xxxx.x runs on Windows 7.

**Comment by Simon-Laux** [comment]
> However, recently I saw that Deltachat has Tauri builds available

Those tauri builds are experimental. I bet that if it even opens, then not all features will work - it was not designed with windows 7 in mind (features like notifications use the newer windows notification api directly instead of using tauri's notification module which has a fallback for win7).
Also tauri itself may not support windows 7 for long, as there is a discussion about dropping support: https://github.com/tauri-apps/tauri/issues/12550
